lib/flight_job/commands/create_script.rb

```diff
@@ -55,17 +97,39 @@ def template
         end
       end
 
-      def stdin_answers
-        # TODO: Validate the correct answers have been provided
-        input = $stdin.read_nonblock(MAX_STDIN_SIZE)
-        if input.length == MAX_STDIN_SIZE
-          raise InputError, "The STDIN exceeds the maximum size of: #{MAX_STDIN_SIZE}B"
+      def stdin_answers?
+        opts.stdin || opts.answers == '@-'
+      end
+
+      def notes_input
+        return unless opts.notes
+        if opts.notes == '@-'
+          cached_stdin
+        elsif opts.notes[0] == '@'
+          read_file(opts.notes[1..])
+        else
+          opts.notes
+        end
+      end
+
+      def answers_input
+        return unless opts.stdin || opts.answers
+        string = if stdin_answers?
+                   cached_stdin
+                 elsif opts.answers[0] == '@'
+                   read_file(opts.answers[1..])
+                 else
+                   opts.answers
+                 end
+        JSON.parse(string).tap do |hash|
+          next if hash.is_a? Hash
+          raise InputError, 'The answers are not a JSON hash'
```

Comment:
Would `raise InputError, 'The answers are not a JSON hash' unless hash.is_a?(Hash)` not be clearer?

Reply:
https://github.com/openflighthpc/flight-job/pull/5/commits/3b7e976634cbdffd8a81b72cce8708f35b02d3e0